src/AspNet.Security.OAuth.Weixin/WeixinAuthenticationHandler.cs

```diff
@@ -109,13 +126,20 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync(string code,
 
         protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
         {
+            var oauthstate = Options.StateDataFormat.Protect(properties);
+            if (Options.AuthorizationEndpoint != WeixinAuthenticationDefaults.AuthorizationEndpoint)
+            {
+                //Store state in redirectUri when authorizing Wechat Web pages to prevent "too long state parameters" error
+                redirectUri = redirectUri.Contains("?") ? $"{redirectUri}&{nameof(oauthstate)}={oauthstate}" : $"{redirectUri}?{nameof(oauthstate)}={oauthstate}";
```

Comment:
Relying on the name of a local variable for a parameter name is brittle against future refactoring. Use a constant.